// containers/registration/index.js
import React, {Component} from 'react';
import {firebaseApp} from 'firebaseConfig';
import {browserHistory} from 'react-router';

class Registration extends Component {
	constructor(){
		super();
		this.state={
			email: '',
			password: '',
			error: {}
		};
		this.handleForm=this.handleForm.bind(this);
	}
	handleForm(e){
		const {email,password} = this.state;
		e.preventDefault();
		let _that=this;
		if(email.length && password.length){
			firebaseApp.auth().createUserWithEmailAndPassword(email, password).catch(function(error) {
				_that.setState({error});
			});
			firebaseApp.auth().onAuthStateChanged(function(user) {
				if (user) {
					browserHistory.push('/');
				}
			});
		}
	}
	render(){
		return(
			<div className='container' style={{marginTop: '20px'}}>
				<div className="row">
					<div className="col-md-12">
						<form onSubmit={this.handleForm}>
							<div className="form-group">
								<label htmlFor="exampleInputLogin">Email</label>
								<input
									type="email"
									className="form-control"
									id="exampleInputLogin"
									value={this.state.email}
									aria-describedby="emailHelp"
									placeholder="Email"
									onChange={(e)=>this.setState({email: e.target.value})}
								/>
							</div>
							<div className="form-group">
								<label htmlFor="exampleInputPassword1">Password</label>
								<input
									type="password"
									className="form-control"
									id="exampleInputPassword1"
									placeholder="Password"
									value={this.props.password}
									onChange={(e)=>this.setState({password: e.target.value})}
								/>
								{this.state.error.message? <p>{this.state.error.message}</p> : '' }
							</div>
							<button
								type="submit"
								className="btn btn-primary"
							>Отправить</button>
						</form>
					</div>
				</div>
			</div>
		)
	}
}

export default Registration;
